UnitOfWork/Services/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.EntityFrameworkCore;
+using UnitOfWork.Data;
+using UnitOfWork.Services.IRepositories;
+
+namespace UnitOfWork.Services.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class
+{
+    protected DataContext _context;
+    protected DbSet<T> dbSet;
+    protected readonly ILogger _logger;
+    
+    public GenericRepository(DataContext context, ILogger logger)
+    {
+        _context = context;
+        this.dbSet = context.Set<T>();
+        _logger = logger;
+    }
+    
+    public virtual async Task<IEnumerable<T>> GetAll()
+    {
+        return await dbSet.ToListAsync();
+    }
+
+    public virtual async Task<T> GetById(int id)
+    {
+        try
+        {
+            return await dbSet.FindAsync(id);
```

Comment:
This code could throw an exception when entity with an id doesn't exist. Better to use `FirstOrDefault` method, and `Find` is good option only when we're sure, that the entity exists.

Reply:
fixed